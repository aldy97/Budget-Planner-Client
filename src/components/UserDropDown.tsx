import React from "react";
import { UserOutlined, LogoutOutlined } from "@ant-design/icons";
import { Menu, Dropdown, Button } from "antd";

interface DropDown {
  name: string;
}

function UserDropDown({ name }: DropDown): JSX.Element {
  const handleLogOutBtnClcik = () => {
    window.location.href = "/";
  };

  const menu = (
    <Menu>
      <Menu.Item>
        <a style={{ textAlign: "center" }} onClick={handleLogOutBtnClcik}>
          <LogoutOutlined style={{ color: "red" }} />
          <span>Log Out</span>
        </a>
      </Menu.Item>
    </Menu>
  );
  return (
    <Dropdown overlay={menu} placement="bottomCenter">
      <Button>
        <UserOutlined></UserOutlined>
        <span data-test="name">{name}</span>
      </Button>
    </Dropdown>
  );
}

export default UserDropDown;
