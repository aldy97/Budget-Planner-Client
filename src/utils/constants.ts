export const COLORS = {
  THEMEBLUE: "#1890ff",
  UNSELECTEDOPTION: "#000000a6",
  ORANGE: "#ffa940",
};

export const ExpenseCategories = [
  "Commute",
  "Meals",
  "Shoppings",
  "Grocery",
  "Veggies",
  "Fruits",
  "Snacks",
  "Sports",
  "Entertainment",
  "Phone Plan",
  "Make up",
  "Housing",
  "Clothing",
  "Children",
  "Elders",
  "Social",
  "Travelling",
  "Wine",
  "Digitals",
  "Car",
  "Medical Care",
  "Books",
  "Learning",
  "Pets",
  "Gifts",
  "Office",
  "Maintainance",
  "Donation",
];

export const IncomeCategories = [
  "Salary",
  "Part-time",
  "Investment",
  "Gifts",
  "Others",
];

export const URL = "http://localhost:7001";
