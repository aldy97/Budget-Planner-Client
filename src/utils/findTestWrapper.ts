import { ReactWrapper } from "enzyme";

export const findTestWrapper = (
  wrapper: ReactWrapper,
  tag: string
): ReactWrapper => {
  return wrapper.find(`[data-test="${tag}"]`);
};
