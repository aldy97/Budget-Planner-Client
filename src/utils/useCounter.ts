import { useEffect, useState } from "react";

export default (): number => {
  const [count, setCount] = useState(0);

  useEffect(() => {
    setTimeout(() => {
      setCount(count + 1);
    }, 1000);
  }, [count]);

  return count;
};
