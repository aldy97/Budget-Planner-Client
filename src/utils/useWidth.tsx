import { useState, useEffect } from "react";

const useInnerWidth = (): number => {
  const [width, setWidth] = useState<number>(window.innerWidth);

  useEffect(() => {
    const handleResize = () => {
      setWidth(window.innerWidth);
    };
    window.addEventListener("resize", handleResize);
  }, [width]);

  return width;
};

export default useInnerWidth;
