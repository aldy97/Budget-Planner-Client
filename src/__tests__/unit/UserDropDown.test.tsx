import React from "react";
import { mount } from "enzyme";
import UserDropDown from "../../components/UserDropDown";
import { findTestWrapper } from "../../utils/findTestWrapper";

describe("UserDropDown", () => {
  const testName = "test";
  const wrapper = mount(<UserDropDown name={testName}></UserDropDown>);

  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("can render name properly", () => {
    const name = findTestWrapper(wrapper, "name");
    expect(name.text()).toEqual(testName);
  });
});
