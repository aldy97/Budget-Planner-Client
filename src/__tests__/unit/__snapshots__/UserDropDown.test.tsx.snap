// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDropDown renders without crashing 1`] = `ReactWrapper {}`;
